### PR TITLE
add usages of data structures to documentation

### PR DESCRIPTION
**Pull request by furkansenharputlu**
This PR adds the usages of remaining data structures. #340

### CODE REVIEW

**Comment by furkansenharputlu** [comment]
verify

**Comment by Serdaro** [comment]
Are there any notes/exceptional situations/gotchas/important-to-knows for these data structures' usages in the Node.js client?

**Comment by mustafaiman** [comment]
@Serdaro @furkansenharputlu I dont think there is any difference worth mentioning. I only have one suggestion: Add one sentence on top of each structure to explain what they are. You can take these from Hazelcast Reference Manual. For instance flake id can be explained like this: `Hazelcast Flake ID Generator is used to generate cluster-wide unique identifiers. Generated identifiers are long primitive values and are k-ordered (roughly ordered). IDs are in the range from 0 to Long.MAX_VALUE.`

File: README.md [hunk 1]
```diff
@@ -1537,6 +1554,174 @@ queue.offer('Furkan').then(function () {
...
+    topic.addMessageListener(function (message) {
+        console.log(message.messageObject);
+    });
+    return topic.publish('Hello to distributed world!');
+});
+```
+
+## 7.4.9 Using Lock
+
+Hazelcast Lock(`ILock`) is a distributed lock implementation. You can synchronize hazelcast members and clients using a Lock. For details refer to [Lock section](https://docs.hazelcast.org/docs/latest/manual/html-single/index.html#lock) in the Hazelcast IMDG Reference Manual.
```
Comment (Serdaro) [inline README.md]: ... Hazelcast members ...

File: README.md [hunk 2]
```diff
@@ -1537,6 +1554,174 @@ queue.offer('Furkan').then(function () {
...
+}).then(function (res) {
+    console.log(res); // 5
+});
+```
+
+## 7.4.12 Using PN Counter
+
+Hazelcast `PNCounter` (Positive-Negative Counter) is a distributed counter implementation where each Hazelcast instance can increment and decrement the counter value and these updates are propagated to all replicas. For details refer to [PN Counter section](https://docs.hazelcast.org/docs/latest/manual/html-single/index.html#pn-counter) in the Hazelcast IMDG Reference Manual.
+
+A CRDT Counter usage example is shown below.
```
Comment (Serdaro) [inline README.md]: A PN Counter usage ...

File: README.md [hunk 3]
```diff
@@ -1537,6 +1554,174 @@ queue.offer('Furkan').then(function () {
...
+    topic.addMessageListener(function (message) {
+        console.log(message.messageObject);
+    });
+    return topic.publish('Hello to distributed world!');
+});
+```
+
+## 7.4.9 Using Lock
+
+Hazelcast Lock(`ILock`) is a distributed lock implementation. You can synchronize Hazelcast Members and clients using a Lock. For details refer to [Lock section](https://docs.hazelcast.org/docs/latest/manual/html-single/index.html#lock) in the Hazelcast IMDG Reference Manual.
```
[the remark this answers is not in the document]
Comment (furkansenharputlu) [reply README.md]: @Serdaro requested that change. as `Hazelcast Members`

File: README.md [hunk 4]
```diff
@@ -1537,6 +1554,174 @@ queue.offer('Furkan').then(function () {
...
+
+Hazelcast List(`IList`) is distributed list which allows duplicate elements and preserves the order of elements. For details refer to [List section](https://docs.hazelcast.org/docs/latest/manual/html-single/index.html#list) in the Hazelcast IMDG Reference Manual.
+
+A List usage example is shown below.
+
+```javascript
+var list;
+hazelcastClient.getList('myList').then(function (l) {
+    list = l;
+    return list.add('Mali');
```
[the remark this answers is not in the document]
Comment (furkansenharputlu) [reply README.md]: It is an abbreviation like Tomas-> Tom :)
But I will change it.

**Comment by furkansenharputlu** [comment]
verify

**Comment by furkansenharputlu** [comment]
verify

**Comment by furkansenharputlu** [comment]
verify

**Pull request merged**